docs/content/focusTrap.mdx

```diff
@@ -0,0 +1,98 @@
+---
+title: Focus Trap Behavior
+---
+
+The `focusTrap` behavior and `useFocusTrap` hook are used prevent focus from leaving a particular element. This is useful for implementing modal dialogs: only the content within the dialog should be interactive, even though the UI underneath may still be visible.
+
+### Behavior
+* Activation: As soon as the focus trap is activated, it will ensure that an element within the container has focus. If it doesn't, it will focus the first focusable element within the container, or, if provided, the element indicated by the `initialFocus` parameter (see API below).
+* External focus changes: If an external cause (e.g. mouse click, scripts, or accessibility software) results in an element outside the container to be focused, focus will immediately be redirected to the last-focused element that is inside the container.
+* Circular tab focus: Using the `TAB` key on the last focusable element within the container will result in the first focusable element within the container receiving focus. Similarly, `Shift+TAB` can be used to focus the last element from the first.
+* Global: Only one focus trap can be _active_ at a time. When a focus trap is enabled, if there is already an active focus trap, it becomes suspended and pushed onto a stack. Once the newly-active focus trap is disabled, the most recently-suspended trap will reactivate. Suspended focus traps can be disabled, causing them to be removed from the stack of suspended traps.
+
+### Demo
+
+To see a demo, deploy Storybook and find the `useFocusTrap` stories.
+
+### Usage
+
+```ts
+function showDialog() {
+  const dialog = document.getElementById("myDialog")
+  if (dialog instanceof HTMLElement) {
+    dialog.style.display = ""
+    return focusTrap(dialog)
+  }
+}
+function hideDialog(controller: AbortController) {
+  document.getElementById("myDialog")?.style.display = "none"
+  controller.abort()
+}
+const dialogController = showDialog()
+
+// later
+if (dialogController) {
+  hideDialog(controller)
+}
+```
+
+### API
+
+The `focusTrap` function takes the following arguments.
+
+| Name | Type | Default | Description |
+| :- | :- | :-: | :- |
+| container | `HTMLElement` | | When active, only elements within this container (along with the container itself) can be focused. |
+| initialFocus | `HTMLElement` | | Specifies the element which will receive focus when the focus trap is activated. Defaults to the first tabbable element inside the container. |
```

Comment:
Set the "Default" cell to _See description_.

Reply:
No problem, this isn't really a full hook. Might be able to refactor this to use yours 👍